src/parse/validator.rs

```diff
@@ -587,31 +587,59 @@ impl<'b, 'c, 'z> Validator<'b, 'c, 'z> {
             .filter(|a| !matcher.contains(a.id))
         {
             debugln!("Validator::validate_required_unless:iter:{};", a.name);
-            if self.fails_arg_required_unless(a, matcher) {
+            if self.fails_arg_required_unless(
+                a.name,
+                a.r_unless.as_ref(),
+                a.is_set(ArgSettings::RequiredUnlessAll),
+                matcher,
+            ) {
                 return self.missing_required_error(matcher, Some(a.id));
             }
         }
 
+        for ag in self
+            .p
+            .app
+            .groups
+            .iter()
+            .filter(|ag| ag.r_unless.is_some())
+            .filter(|ag| ag.args.iter().all(|&arg_id| !matcher.contains(arg_id)))
+        {
+            debugln!(
+                "Validator::validate_required_unless:arg_group_iter:{};",
+                ag.name
+            );
+            if self.fails_arg_required_unless(ag.name, ag.r_unless.as_ref(), false, matcher) {
+                return self.missing_required_error(matcher, None);
+            }
+        }
+
         Ok(())
     }
 
     // Failing a required unless means, the arg's "unless" wasn't present, and neither were they
-    fn fails_arg_required_unless(&self, a: &Arg<'b>, matcher: &ArgMatcher) -> bool {
-        debugln!("Validator::fails_arg_required_unless: a={:?};", a.name);
+    fn fails_arg_required_unless(
+        &self,
+        _name: &str,
```

Comment:
Despite `_name` definitely being used here, `cargo check` complains if it's called `name` since it believes the variable isn't used. My guess is because it's only used as part of a macro call

Reply:
Yeah, cargo check is not perfect.

Reply:
I reported it to clippy https://github.com/rust-lang/rust-clippy/issues/5472 . Meanwhile, I'm going to tweak `debugln!` to ignore this specific lint.

Reply:
As answered in that thread, this is a feature, not a bug.